portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,18 +38,29 @@
   // Returns comments in JSON format
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String lang = request.getParameter("lang");
     Query query = new Query("Comment").addSort("timestampMs", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
+    
+    Translate translate = TranslateOptions.getDefaultInstance().getService();
 
     ArrayList<String> comments = new ArrayList<>();
     for (Entity entity : results.asIterable()) {
       String comment = (String) entity.getProperty("text");
-      comments.add(comment);
+      if(lang.length()>0){
+        Translation translation =
+        translate.translate(comment, Translate.TranslateOption.targetLanguage(lang));
```

Comment:
nit: indent a couple spaces to make it clear it is a continuation from the previous line.